tests/integration/internal/receivermock/receiver_mock.go

```diff
@@ -20,10 +26,32 @@ type ReceiverMockClient struct {
 	tlsConfig tls.Config
 }
 
-func NewReceiverMockClient(t *testing.T, baseUrl url.URL) *ReceiverMockClient {
+func NewClient(t *testing.T, baseUrl url.URL) *ReceiverMockClient {
 	return &ReceiverMockClient{baseUrl: baseUrl, tlsConfig: tls.Config{}}
 }
 
+// NewClientWithK8sTunnel creates a client for receiver-mock.
+// It return the client itself and a tunnel teardown func which should be called
+// by the caller when they're done with it.
+func NewClientWithK8sTunnel(
+	ctx context.Context,
+	t *testing.T,
+) (*ReceiverMockClient, func()) {
+	tunnel := k8s.TunnelForReceiverMock(ctx, t)
+	baseUrl := url.URL{
+		Scheme: "http",
+		Host:   tunnel.Endpoint(),
+		Path:   "/",
+	}
+
+	return &ReceiverMockClient{
+			baseUrl:   baseUrl,
+			tlsConfig: tls.Config{},
+		}, func() {
+			tunnel.Close()
+		}
+}
+
 func (client *ReceiverMockClient) GetMetricCounts(t *testing.T) (MetricCounts, error) {
 	path, err := url.Parse("metrics-list")
 	if err != nil {
```

Comment:
Actually it's good you're pointing this out. I've removed the used of `require` in here because I don't want the test to fail when that func is called inside of `assert.Eventually` and it succeeds before timing out.